include/bfs.h

```diff
@@ -1,15 +1,15 @@
-// Copyright 2024 Aditya Behrani
-#ifndef BFS_H
-#define BFS_H
-
-#include <vector>
-#include "../include/graph.h"
-
-class BFS {
-    public:
-        static std::vector<int>* bfs(
-            Graph *graph,
-            int start,
-            int goal);
-};
```

Comment:
C++ supports object oriented programming. Why not write each pathfinder algorithm as a subclass (implementation) of an interface?